src/researches/findKeywordInFirstParagraph.js

```diff
@@ -1,23 +1,83 @@
- * if there is no paragraph tag or 0 hits, it checks for 2 newlines, otherwise returns the keyword
- * count of the complete text.
+ * Checks if the paragraph consists only of images.
+ *
+ * @param {string} text The text string to analyze.
+ *
+ * @returns {boolean} True if the text consists only of images, false otherwise.
+ */
+function paragraphConsistsOfImagesOnly( text ) {
+	const images = imageInText( text );
+	if ( images.length < 1 ) {
+		return false;
+	}
+
+	images.forEach( function( image ) {
+		text = text.replace( image, "" );
+	} );
+
+	return text === "";
+}
+
+/**
+ * First splits the first paragraph by sentences. Finds the first paragraph which contains sentences e.g., not an image).
+ * Tries to find all (content) words from the keyphrase or a synonym phrase within one sentence.
+ *    If found all words within one sentence, returns an object with foundInOneSentence = true and
+ *                                                                         keyphraseOrSynonym = "keyphrase" or "synonym".
+ *    Else goes ahead with matching then entire first paragraph.
+ * Tries to find all (content) words from the keyphrase or a synonym phrase within the paragraph.
+ *    If found all words within the paragraph, returns an object with foundInOneSentence = false,
+ *                                            foundInParagraph = true, and keyphraseOrSynonym = "keyphrase" or "synonym".
+ *    If found not all words within the paragraph of nothing at all, returns an object with foundInOneSentence = false,
+ *                                                                 foundInParagraph = false, and keyphraseOrSynonym = "".
  *
  * @param {Paper} paper The text to check for paragraphs.
- * @returns {number} The number of occurrences of the keyword in the first paragraph.
+ * @returns {Object} Whether the keyphrase words were found in one sentence, whether the keyphrase words were found in
+ * the paragraph, whether a keyphrase or a synonym phrase was matched.
  */
-module.exports = function( paper ) {
-	const paragraphs = matchParagraphs( paper.getText() );
-	const keyword = escapeRegExp( paper.getKeyword().toLocaleLowerCase() );
-	const paragraph = reject( paragraphs, isEmpty )[ 0 ] || "";
-	return wordMatch( paragraph, keyword, paper.getLocale() ).count;
-};
+export default function( paper ) {
+	const topicForms = paper.getTopicForms();
+	const locale = paper.getLocale();
+
+	let paragraphs = matchParagraphs( paper.getText() );
+	paragraphs = reject( paragraphs, isEmpty );
+
+	let paragraph = reject( paragraphs, paragraphConsistsOfImagesOnly )[ 0 ] || "";
+
+	let result = {
+		foundInOneSentence: false,
+		foundInParagraph: false,
+		keyphraseOrSynonym: "",
+	};
+
+	const sentences = getSentences( paragraph );
+	if ( ! isEmpty( sentences ) ) {
+		sentences.forEach( function( sentence ) {
+			const resultSentence = findTopicFormsInString( topicForms, sentence, true, locale );
```

Comment:
About this `true`. I think it would be nicer if this was not hardcoded in two spots here. This would allow for naming it too.
For example: `const useSynonyms = true;`.